MediaTracker/app/src/main/java/com/izor066/android/mediatracker/ui/AddNewEntryManually.java

```diff
@@ -209,19 +218,91 @@ public boolean onCreateOptionsMenu(Menu menu) {
         return true;
     }
 
-    private void loadCover() {
-        if (addCover.getText().length() == 0) {
-            mAddCover = IMAGE_PLACEHOLDER;
+    private void captureCoverFromGallery() {
+
+        Intent pickPictureIntent = new Intent(Intent.ACTION_PICK, android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+        if (pickPictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(pickPictureIntent, RESULT_LOAD_IMAGE);
+        }
+
+    }
+
+    private void captureCoverWithCamera() {
+
+        Intent takePictureIntent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+        if (takePictureIntent.resolveActivity(getPackageManager()) != null) {
+            startActivityForResult(takePictureIntent, REQUEST_IMAGE_CAPTURE);
+        }
+
+
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if (requestCode == RESULT_LOAD_IMAGE && resultCode == RESULT_OK && null != data) {
+            Uri selectedImage = data.getData();
+            String[] filePathColumn = {MediaStore.Images.Media.DATA};
+
+            Cursor cursor = getContentResolver().query(selectedImage, filePathColumn, null, null, null);
+            cursor.moveToFirst();
+
+            int columnIndex = cursor.getColumnIndex(filePathColumn[0]);
+            mAddCover = cursor.getString(columnIndex);
+            cursor.close();
+
+            Log.e(TAG, mAddCover);
+
+            coverPreview.setVisibility(View.VISIBLE);
+            Picasso.with(this)
+                    .load(new File(mAddCover))
+                    .into(coverPreview);
+            isImgLoaded = true;
+        }
+
+        if (requestCode == REQUEST_IMAGE_CAPTURE && resultCode == RESULT_OK) {
 
-        } // has to be in, otherwise the placeholder won't load if the user enters a non-valid url and then deletes it
+            Bundle extras = data.getExtras();
+            Bitmap bitmapRaw = (Bitmap) extras.get("data");
+            Bitmap imageBitmap = rotateBitmap(bitmapRaw, 90);
+
+            ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+            imageBitmap.compress(Bitmap.CompressFormat.PNG, 100, bytes);
+            File destination = new File(Environment.getExternalStorageDirectory(), System.currentTimeMillis() + ".jpg");
+
+            FileOutputStream fo;
+            try {
+                destination.createNewFile();
+                fo = new FileOutputStream(destination);
+                fo.write(bytes.toByteArray());
+                fo.close();
+            } catch (FileNotFoundException e) {
+                e.printStackTrace();
```

Comment:
These should be `Log.e()` not `printStackTrace()`, the later is rather expensive and not that useful.